http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -58,31 +62,89 @@ protected void channelRead0(ChannelHandlerContext channelContext, HttpObject msg
         RequestContext requestContext = channelContext.channel().attr(REQUEST_CONTEXT_KEY).get();
 
 
-        if (msg instanceof StreamedHttpResponse) {
-            StreamedHttpResponse response = (StreamedHttpResponse) msg;
+        if (msg instanceof HttpResponse) {
+            HttpResponse response = (HttpResponse) msg;
             SdkHttpResponse sdkResponse = SdkHttpFullResponse.builder()
                                                              .headers(fromNettyHeaders(response.headers()))
                                                              .statusCode(response.status().code())
                                                              .statusText(response.status().reasonPhrase())
                                                              .build();
             channelContext.channel().attr(KEEP_ALIVE).set(HttpUtil.isKeepAlive(response));
             requestContext.handler().headersReceived(sdkResponse);
-            requestContext.handler().onStream(new PublisherAdapter(response, channelContext, requestContext));
+        }
+
+        if (msg instanceof StreamedHttpResponse) {
+            requestContext.handler().onStream(new PublisherAdapter((StreamedHttpResponse) msg, channelContext, requestContext));
+        } else if (msg instanceof FullHttpResponse) {
```

Comment:
Does this fix all of the remaining empty-response integration test failures?

Reply:
Yeah, should fix them.